benchmark/src/main/kotlin/com/github/avrokotlin/benchmark/internal/ListWrapperDataClass.kt

```diff
@@ -0,0 +1,33 @@
+package com.github.avrokotlin.benchmark.internal
+
+import kotlinx.serialization.Serializable
+
+@Serializable
+internal data class ListWrapperDataClass(val index: Int, val entries: List<StatsEntry>) {
+    companion object {
+        fun create(size: Int, entriesSize: Int, random: RandomUtils = RandomUtils()) =
+            ListWrapperDataClass(random.nextInt(), List(size) { StatsEntry.create(entriesSize) })
+    }
+}
+
+@Serializable
+internal data class StatsEntry(val elementId: ElementId, val values: List<StatValue>) {
+    companion object {
+        fun create(entriesSize: Int, random: RandomUtils = RandomUtils()) =
+            StatsEntry(random.nextLong(), List(entriesSize) { random.nextLong() })
+    }
+}
+
+typealias ElementId = Long
+typealias StatValue = Long
```

Comment:
use internal everywhere to not change the apiDump

```suggestion
internal typealias ElementId = Long
internal typealias StatValue = Long
```

Reply:
I added these type aliases here to make sure that it doesn't affect anything because I have a similar problem in the code where I originally noticed the problem. I wonder if I should leave it as is or simplify it (eg first version with just list of ints 
 which I think shows the issue without any additional complexity needed)?

Reply:
No keep both, but you can remove the typealias, finally the main issue is the low-level encoding/decoding of ints. I'll also rename your issue and PR